File: quiz/programers/js/level2/기능개발.js
function solution(progresses, speeds) {
  const answer = [];
  while (true) {
    let count = 0;
    for (let i = 0; i < progresses.length; i++) {
      progresses[i] += progresses[i] + speeds[i] <= 100 ? speeds[i] : 100 - progresses[i];
    }
    if (progresses[0] === 100) {
      while (progresses[0] === 100) {
        progresses.shift();
        speeds.shift();
        count++;
      }
      answer.push(count);
    }
    if (progresses.length === 0) {
      break;
    }
  }
  return answer;
}
console.log(solution([93, 30, 55], [1, 30, 5]));

File: quiz/programers/python/level2/기능개발.py
def solution(progresses, speeds):
    answer = []
    while progresses:
        count = 0
        for i in range(len(progresses)):
            progresses[i] += speeds[i]
        print(progresses)
        while True and len(progresses):
            if progresses[0] < 100:
                break
            progresses.pop(0)
            speeds.pop(0)
            count += 1
        if count:
            answer.append(count)
    return answer


print(solution([93, 30, 55], [1, 30, 5]))
